src/D7/FileDownloadTrait.php

```diff
@@ -187,7 +191,7 @@ protected function fileDownloadOpenZip() {
       throw new \Exception('Downloaded file information does not have content type data.');
     }
 
-    if ($this->fileDownloadDownloadedFileInfo['content_type'] != 'application/octet-stream') {
+    if ($this->fileDownloadDownloadedFileInfo['content_type'] == 'application/octet-stream') {
```

Comment:
Why we should check a zip header?

Reply:
sorry, I do not understand the question.

This is a ZIP-specific step, and `application/octet-stream` is a correct header for ZIP, unless I'm wrong  - please provide a link if I am

Reply:
Simple .zip hasn't this header. It has "application/zip". Anyway, I can't create archive with this header.

Reply:
oh, ok, can you please change this to accommodate both headers and if neither of them is present - throw exception.